kafka-plugins-client/src/test/java/io/cdap/plugin/kafka/KafkaSinkAndAlertsPublisherTest.java

```diff
@@ -228,7 +228,86 @@ public Alert apply(String s) {
+      "user",
+      Schema.Field.of("id", Schema.nullableOf(Schema.of(Schema.Type.LONG))),
+      Schema.Field.of("first", Schema.of(Schema.Type.STRING)),
+      Schema.Field.of("last", Schema.of(Schema.Type.NULL)));
+
+    // create the pipeline config
+    String inputName = "sinkTestInput";
+
+    String usersTopic = "records";
+    String alertsTopic = "alerts";
+    Map<String, String> sinkProperties = new HashMap<>();
+    sinkProperties.put("brokers", "localhost:" + kafkaPort);
+    sinkProperties.put("referenceName", "kafkaTest");
+    sinkProperties.put("topic", usersTopic);
+    sinkProperties.put("schema", schema.toString());
+    sinkProperties.put("format", "csv");
+    sinkProperties.put("key", "last");
+    sinkProperties.put("async", "FALSE");
+    sinkProperties.put("compressionType", "none");
+
+    Map<String, String> alertProperties = new HashMap<>();
+    alertProperties.put("brokers", "localhost:" + kafkaPort);
+    alertProperties.put("topic", alertsTopic);
+
+    ETLStage source = new ETLStage("source", MockSource.getPlugin(inputName));
+    ETLStage sink =
+      new ETLStage("sink", new ETLPlugin("Kafka", KafkaBatchSink.PLUGIN_TYPE, sinkProperties, null));
+
+    ETLBatchConfig pipelineConfig = ETLBatchConfig.builder("* * * * *")
+      .addStage(source)
+      .addStage(sink)
+      .addConnection(source.getName(), sink.getName())
+      .build();
+
+    // create the pipeline
+    ApplicationId pipelineId = NamespaceId.DEFAULT.app("testKafkaSink");
+    ApplicationManager appManager = deployApplication(pipelineId, new AppRequest<>(APP_ARTIFACT, pipelineConfig));
+    
+    Set<String> expected = ImmutableSet.of("100,samuel,jackson",
+                                           "200,dwayne,johnson",
+                                           "300,christopher,walken",
+                                           "400,donald,trump");
+
+    List<StructuredRecord> records = new ArrayList<>();
+    for (String e : expected) {
+      String[] splits = e.split(",");
+      StructuredRecord record =
+        StructuredRecord.builder(schema)
+          .set("id", Long.parseLong(splits[0]))
+          .set("first", splits[1])
+          .set("last", splits[2])
+          .build();
+      records.add(record);
+    }
+
+    // Add a null record to get an alert
+    StructuredRecord nullRecord =
+      StructuredRecord.builder(schema)
+        .set("id", 500L)
+        .set("first", "terry")
+        .set("last", null)
+        .build();
+    records.add(nullRecord);
+
+    DataSetManager<Table> sourceTable = getDataset(inputName);
+    MockSource.writeInput(sourceTable, records);
+
+    WorkflowManager workflowManager = appManager.getWorkflowManager(SmartWorkflow.NAME);
+    try {
+      workflowManager.start();
+      workflowManager.waitForRun(ProgramRunStatus.COMPLETED, 1, TimeUnit.MINUTES);
+    } catch (Exception e) {
+      Assert.assertTrue(workflowManager.getHistory(ProgramRunStatus.FAILED).size() == 1);
```

Comment:
Also add a success case that uses `key` field.

Reply:
The success case already exists in the previous unit test. `KafkaSinkAndAlertsPublisherTest.testKafkaSinkAndAlertsPublisher()`